lib/models/settings_model.dart

```diff
@@ -90,11 +91,26 @@ class SettingsModel implements Model {
   /// List of weekday colors shown in the weekplan
   List<WeekdayColorModel> weekDayColors;
 
+  /// Get the [WeekdayColorModel] for the given [Weekday].
+  WeekdayColorModel getWeekdayColor(Weekday weekDay) {
+    int i = weekDay.index;
+    do {
```

Comment:
Perhaps use a "for-loop" instead if you use an "int i" anyways?

Reply:
It was intentionally done this way to allow the looping to start at the index which is correct most/all of the time. Using a for-loop would require some weird hack to allow it to start at `weekDay.index` and end at `(weekDay.index - 1) % weekDayColors.length`